### PR TITLE
Docs: Update Getting started and docs clean-up

### PR DESCRIPTION
**Pull request by mbevc1**
**1. Issue, if available:**
n/a

**2. Description of changes:**
* Update documentation and add Terraform example using Fargate as compute backend.
* Clean up some empty spaces and obsolete text

**3. How was this change tested?**
n/a

**4. Does this change impact docs?**
- [x] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by geoffcline** [comment]
thanks and welcome! :)

we have previously declined to include fargate in the getting started guide. reasons include "mixed messaging" on node mgmt and introduction of an aws proprietary concept into cluster operations. 

Can you (A) remove fargate from the getting started guide, (B) add a comment explaining why it's important to include, (C) move the fargate content to a different area (e.g., new task)? 

Thanks! Geoffrey

File: website/content/en/docs/getting-started-with-terraform/_index.md [hunk 1]
```diff
@@ -109,6 +116,48 @@ module "eks" {
...
+      ]
+
+      tags = {
+        owner = "karpenter"
+      }
+    },
+  }
+}
+```
+
```
Comment (geoffcline) [inline website/content/en/docs/getting-started-with-terraform/_index.md:160]: see previous comment

**Review by geoffcline — CHANGES_REQUESTED** [review]

**Comment by mbevc1** [comment]
Hi @geoffcline!

Thanks for your feedback. I believe having Fargate as an deployment howto is a valid option since it currently only works on AWS anyways and using Fargate for those controllers seems a really good use case :) 

I'm happy to either break it to a separate section or add additional comments to the docs. Please let me know what would be your preference.

**Comment by ellistarn** [comment]
> Hi @geoffcline!
> 
> Thanks for your feedback. I believe having Fargate as an deployment howto is a valid option since it currently only works on AWS anyways and using Fargate for those controllers seems a really good use case :)
> 
> I'm happy to either break it to a separate section or add additional comments to the docs. Please let me know what would be your preference.

One of the things we're worried about is docs sprawl. Given the amount of resources we have to keep the docs accurate and understandable, we need to be careful about how much content is in scope. At the end of the day, there are so many ways to configure kubernetes on aws (fargate,mng,kops) as well as ways to configure addons (irsa/k2iam, cnis, etc), that we can't hope to get them all. Instead, the goal is to document ~1 happy path that demonstrates the components involved, and customers can swap in alternatives as makes sense for their use case. 

I'd love to see this kind of content in a medium blog, which we will happily link to on the main README.

**Comment by mbevc1** [comment]
> > Hi @geoffcline!
> > Thanks for your feedback. I believe having Fargate as an deployment howto is a valid option since it currently only works on AWS anyways and using Fargate for those controllers seems a really good use case :)
> > I'm happy to either break it to a separate section or add additional comments to the docs. Please let me know what would be your preference.
> 
> One of the things we're worried about is docs sprawl. Given the amount of resources we have to keep the docs accurate and understandable, we need to be careful about how much content is in scope. At the end of the day, there are so many ways to configure kubernetes on aws (fargate,mng,kops) as well as ways to configure addons (irsa/k2iam, cnis, etc), that we can't hope to get them all. Instead, the goal is to document ~1 happy path that demonstrates the components involved, and customers can swap in alternatives as makes sense for their use case.
> 
> I'd love to see this kind of content in a medium blog, which we will happily link to on the main README.

That makes sense and that is intent as well. I think amount of change here is minimal and it's just adding Fargate option (1 profile with two namespaces) to Terraform path. Maintenance burden is quite low as this bit is not going to change. If you would still prefer to not include it in getting started, I'm happy to link it to my repo.

**Comment by ellistarn** [comment]
Hey @mbevc1, I think we're going to punt on this scope for now. I'd be happy to link to from our main README to "unofficial" docs for Karpenter+Fargate.

**Comment by mbevc1** [comment]
@geoffcline @ellistarn thanks for the feedback. Removed Fargate scenario here and only leaving minor docs updates.

File: website/content/en/v0.5.5/getting-started-with-terraform/_index.md [hunk 2]
```diff
@@ -311,10 +306,6 @@ spec:
       cpu: 1000
   provider:
     instanceProfile: KarpenterNodeInstanceProfile-${CLUSTER_NAME}
-    subnetSelector:
```
Comment (ellistarn) [inline website/content/en/v0.5.5/getting-started-with-terraform/_index.md:314]: These are now required, as of v0.5.5

Comment (mbevc1) [reply website/content/en/v0.5.5/getting-started-with-terraform/_index.md:314]: oh, let me put it back now, miseed it from 0.5.5 release notes :)

**Comment by ellistarn** [comment]
Thanks Marko, just one minor detail left.

File: website/content/en/preview/getting-started-with-terraform/_index.md [hunk 3]
```diff
@@ -106,15 +106,10 @@ module "eks" {
       asg_max_size  = 1
     }
   ]
-
-  tags = {
-    "karpenter.sh/discovery" = var.cluster_name
-  }
```
Comment (ellistarn) [inline website/content/en/preview/getting-started-with-terraform/_index.md:112]: We need this one, too

File: website/content/en/v0.5.5/getting-started-with-terraform/_index.md [hunk 4]
```diff
@@ -106,15 +106,10 @@ module "eks" {
       asg_max_size  = 1
     }
   ]
-
-  tags = {
-    "karpenter.sh/discovery" = var.cluster_name
-  }
```
Comment (ellistarn) [inline website/content/en/v0.5.5/getting-started-with-terraform/_index.md:112]: We need this one

File: website/content/en/preview/getting-started-with-terraform/_index.md (hunk 3 above)
Comment (mbevc1) [reply website/content/en/preview/getting-started-with-terraform/_index.md:112]: :roll_eyes: ouch, seems that came in from rebasing earlier, fixing now!

**Review by ellistarn — APPROVED** [review]
Thanks!

**Comment by mbevc1** [comment]
:wave: @geoffcline all good on your end for merging this one?

**Comment by geoffcline** [comment]
I added a reminder to review this Monday morning :) thanks for the ping

**Comment by geoffcline** [comment]
approved and merged!

in the future you only need to update /preview, no need to update the archive versions.

**Comment by mbevc1** [comment]
Thanks :+1:

**Pull request merged**